homeassistant/components/local_calendar/calendar.py

```diff
@@ -44,7 +44,9 @@ async def async_setup_entry(
     """Set up the local calendar platform."""
     store = hass.data[DOMAIN][config_entry.entry_id]
     ics = await store.async_load()
-    calendar = IcsCalendarStream.calendar_from_ics(ics)
+    calendar: Calendar = await hass.async_add_executor_job(
```

Comment:
I guess it doesn't happen often enough to fix, but if [this is the cause](https://github.com/allenporter/ical/blob/64380cefb6942f75e76d725d1cf01ee39fe63b43/ical/util.py#L30-L32), why not push the call to `metadata` to be stored in a top level constant?

Reply:
I considered moving that one, and may do that.  My assumption was there was also some stuff like https://github.com/allenporter/ical/blob/e46eb7329ac633d7e5002f67b4de57368d6e9007/ical/tzif/timezoneinfo.py#L75 given https://github.com/home-assistant/core/issues/118627 which shares the same code and so i didn't want to keep chasing this.

My take was: Home assistant is being pretty strict here -- so i'm putting the complexity in home assistant. (i get the motivation, it makes sense, just  hard to chase down the root cause since it can't be reproduced locally.